spec/support/shared_examples_for_personal_keys.rb

```diff
@@ -12,4 +12,26 @@
       end
     end
   end
+
+  context 'with javascript enabled', js: true do
+    before do
+      page.driver.browser.execute_cdp(
+        'Browser.grantPermissions',
+        origin: page.server_url,
+        permissions: ['clipboardReadWrite', 'clipboardSanitizedWrite'],
+      )
+    end
+
+    after do
+      page.driver.browser.execute_cdp('Browser.resetPermissions')
+    end
+
+    it 'allows a user to copy the code into the confirmation modal' do
+      click_on t('links.copy')
+      copied_text = page.evaluate_async_script('navigator.clipboard.readText().then(arguments[0])')
```

Comment:
what's `arguments` here?

Reply:
> what's `arguments` here?

From what I understand, `evaluate_async_script` wraps the given script in a function, called with a "callback" function argument ([last argument?](https://www.rubydoc.info/github/jnicklas/capybara/Capybara%2FSession:evaluate_async_script)).

So it ends up being something like...

```js
(function(callback) {
  navigator.clipboard.readText().then(arguments[0])
})(/*...*/);
```

`arguments` being the [JavaScript keyword](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments) to reference the given function arguments.

Reply:
neat, I wish they'd call it it something clearer like `resolve` lol 😂

Reply:
Initially I'd hoped it would infer the promise value and treat it as something more like...

```rb
copied_text = page.evaluate_async_script('navigator.clipboard.readText()')
```

```js
(() => navigator.clipboard.readText())().then(callback);
```